### ISSUE TITLE
Recent changes to Trilinos are causing segfaults when using OpenMP with GCC 7.2.0

### ISSUE THREAD

**Issue by jrood-nrel** [trilinos/Trilinos#2130]
## Steps to Reproduce
Running the regression tests for Nalu (https://my.cdash.org/index.php?project=Nalu) on OSX 10.12.6 using GCC 7.2.0 with OpenMP enabled in the latest (052a35cc for example) Trilinos can reproduce this issue. OpenMP runtime options used are:
```
export OMP_NUM_THREADS=1
export OMP_PROC_BIND=false
```

## Description of Error
60/78 of the Nalu regression tests fail due to segfaults either indicating Kokkos, or a nondescript segfault. This arose sometime between January 2-4 where using an earlier version of Trilinos from Jan 2, or turning off OpenMP do not cause the segfaults.

Below is the listing of the most descriptive error:
```
naluX(54561,0x1133763c0) malloc: *** mach_vm_map(size=231034881392640) failed (error code=3)
*** error: can't allocate region
*** set a breakpoint in malloc_error_break to debug
Kokkos::HostSpace::allocate[ STD_MALLOC ]( 231034881389296 ) FAILED NULL
terminate called after throwing an instance of 'std::runtime_error'
  what():  Kokkos::HostSpace::allocate[ STD_MALLOC ]( 231034881389296 ) FAILED NULL
```

**Comment by mhoemmen** [comment]
(FYI Trilinos only just recently started having a Dashboard build for GCC 7.2.0 on Linux.)

**Comment by mhoemmen** [comment]
@jrood-nrel Would you happen to know a last working Trilinos:develop commit?

**Comment by jrood-nrel** [comment]
@mhoemmen The last working commit for Trilinos I can find regarding this issue with our testing is `9874b9dd Merge remote branch 'intermediate-repo/develop' into develop` which looks to be Jan 2. Sorry I don't have a data point for Jan 3 to know exactly when this showed up though. Somewhere between Jan 2-4 actually. Our tests run at midnight.

**Comment by jrood-nrel** [comment]
@mhoemmen I looked further into this with these results:
```
Neighboring Trilinos develop commits regarding segfaults on OSX with GCC and OpenMP:
841e89f5cb58e7d034ab6c68fb3887b2d39c5f5 bad
bb82440caacd3b154a8bc087f1d081ca39a108f good
1565f43d2a42c7fe92f0f8bd74367a26f0664d1 good
```

**Comment by bartlettroscoe** [comment]
FYI: Looks like it happened again. Two CI builds are running on top of each other on ceerws1113 and causing random configure and build failures in the last two CI iterations with build stamps `20180114-1100-Continuous` and `20180115-1100-Continuous`.  Two jobs are shown running at the same time on the machine:

```
ps -AF | grep trilinos_ci_server.out
rabartl  28494 28493  0 26527  1172   1 04:00 ?        00:00:00 /bin/sh -c cd /scratch/rabartl/Trilinos.base/SEMSCIBuild && cp trilinos_ci_server.out trilinos_ci_server.last.out && ./Trilinos/cmake/ctest/drivers/sems_ci/trilinos_ci_sever.sh &> trilinos_ci_server.out
rabartl  42175 21021  0 25834   956   1 08:41 pts/4    00:00:00 grep trilinos_ci_server.out
rabartl  45160 45156  0 26527  1172  22 Jan14 ?        00:00:00 /bin/sh -c cd /scratch/rabartl/Trilinos.base/SEMSCIBuild && cp trilinos_ci_server.out trilinos_ci_server.last.out && ./Trilinos/cmake/ctest/drivers/sems_ci/trilinos_ci_sever.sh &> trilinos_ci_server.out
```

I will kill the jobs and restart the CI server.   But it is time for a solution so that this does not happen again.

**Comment by jrood-nrel** [comment]
I can reproduce the segfaults with GCC 7.2.0 on Linux with OpenMP on. So it turns out the segfaults are not an OSX issue.

**Comment by bartlettroscoe** [comment]
> I can reproduce the segfaults with GCC 7.2.0 on Linux with OpenMP on. So it turns out the segfaults are not an OSX issue.

We now have a GCC 7.2.0 build running (see #2082).  It looks like OpenMP is **NOT** enabled for that build.  Should it be?

**Comment by jrood-nrel** [comment]
@bartlettroscoe That sounds like a good idea regarding this issue.

**Comment by jhux2** [comment]
@jrood-nrel I've rebuilt on Sandia's CEE lan with GCC 7.2.0, and see this test failure:

```
mpirun -np 6 valgrind /scratch/jhu/nalu/nalu/build-gcc-7.2.0-openmpi-1.10.2/naluX -i /scratch/jhu/nalu/nalu/Nalu/reg_tests/test_files/oversetFluidsEdge/oversetFluidsEdge.i -o oversetFluidsEdge.log
==12924== Memcheck, a memory error detector
==12924== Copyright (C) 2002-2012, and GNU GPL'd, by Julian Seward et al.
==12924== Using Valgrind-3.8.1 and LibVEX; rerun with -h for copyright info
==12924== Command: /scratch/jhu/nalu/nalu/build-gcc-7.2.0-openmpi-1.10.2/naluX -i /scratch/jhu/nalu/nalu/Nalu/reg_tests/test_files/oversetFluidsEdge/oversetFluidsEdge.i -o oversetFluidsEdge.log
==12924== 
==12925== Memcheck, a memory error detector
==12925== Copyright (C) 2002-2012, and GNU GPL'd, by Julian Seward et al.
==12925== Using Valgrind-3.8.1 and LibVEX; rerun with -h for copyright info
==12925== Command: /scratch/jhu/nalu/nalu/build-gcc-7.2.0-openmpi-1.10.2/naluX -i /scratch/jhu/nalu/nalu/Nalu/reg_tests/test_files/oversetFluidsEdge/oversetFluidsEdge.i -o oversetFluidsEdge.log
==12925== 
==12926== Memcheck, a memory error detector
==12926== Copyright (C) 2002-2012, and GNU GPL'd, by Julian Seward et al.
==12926== Using Valgrind-3.8.1 and LibVEX; rerun with -h for copyright info
==12926== Command: /scratch/jhu/nalu/nalu/build-gcc-7.2.0-openmpi-1.10.2/naluX -i /scratch/jhu/nalu/nalu/Nalu/reg_tests/test_files/oversetFluidsEdge/oversetFluidsEdge.i -o oversetFluidsEdge.log
==12926== 
==12928== Memcheck, a memory error detector
==12928== Copyright (C) 2002-2012, and GNU GPL'd, by Julian Seward et al.
==12928== Using Valgrind-3.8.1 and LibVEX; rerun with -h for copyright info
==12928== Command: /scratch/jhu/nalu/nalu/build-gcc-7.2.0-openmpi-1.10.2/naluX -i /scratch/jhu/nalu/nalu/Nalu/reg_tests/test_files/oversetFluidsEdge/oversetFluidsEdge.i -o oversetFluidsEdge.log
==12928== 
==12929== Memcheck, a memory error detector
==12929== Copyright (C) 2002-2012, and GNU GPL'd, by Julian Seward et al.
==12929== Using Valgrind-3.8.1 and LibVEX; rerun with -h for copyright info
==12929== Command: /scratch/jhu/nalu/nalu/build-gcc-7.2.0-openmpi-1.10.2/naluX -i /scratch/jhu/nalu/nalu/Nalu/reg_tests/test_files/oversetFluidsEdge/oversetFluidsEdge.i -o oversetFluidsEdge.log
==12929== 
==12927== Memcheck, a memory error detector
==12927== Copyright (C) 2002-2012, and GNU GPL'd, by Julian Seward et al.
==12927== Using Valgrind-3.8.1 and LibVEX; rerun with -h for copyright info
==12927== Command: /scratch/jhu/nalu/nalu/build-gcc-7.2.0-openmpi-1.10.2/naluX -i /scratch/jhu/nalu/nalu/Nalu/reg_tests/test_files/oversetFluidsEdge/oversetFluidsEdge.i -o oversetFluidsEdge.log
==12927== 
Kokkos::HostSpace::allocate[ STD_MALLOC ]( 1958505096696 ) FAILED NULLKokkos::HostSpace::allocate[ STD_MALLOC ]( 2989297246184 ) FAILED NULLKokkos::HostSpace::allocate[ STD_MALLOC ]( 2302102479032 ) FAILED NULLKokkos::HostSpace::allocate[ STD_MALLOC ]( 2130303789080 ) FAILED NULLKokkos::HostSpace::allocate[ STD_MALLOC ]( 3195455678560 ) FAILED NULLKokkos::HostSpace::allocate[ STD_MALLOC ]( 3264175154992 ) FAILED NULL





terminate called after throwing an instance of 'terminate called after throwing an instance of 'std::runtime_errorterminate called after throwing an instance of ''
std::runtime_errorterminate called after throwing an instance of 'std::runtime_errorterminate called after throwing an instance of ''
terminate called after throwing an instance of ''
std::runtime_errorstd::runtime_error'
std::runtime_error'
'
  what():  Kokkos::HostSpace::allocate[ STD_MALLOC ]( 1958505096696 ) FAILED NULL
Traceback functionality not available
  what():  Kokkos::HostSpace::allocate[ STD_MALLOC ]( 2989297246184 ) FAILED NULL
Traceback functionality not available

  what():  Kokkos::HostSpace::allocate[ STD_MALLOC ]( 2302102479032 ) FAILED NULL
Traceback functionality not available

  what():  Kokkos::HostSpace::allocate[ STD_MALLOC ]( 3195455678560 ) FAILED NULL
Traceback functionality not available

  what():  Kokkos::HostSpace::allocate[ STD_MALLOC ]( 3264175154992 ) FAILED NULL
Traceback functionality not available
  what():  Kokkos::HostSpace::allocate[ STD_MALLOC ]( 2130303789080 ) FAILED NULL
Traceback functionality not available
```

I'm waiting on the results of a pure debug Nalu build -- valgrind is *really* slow.

**Comment by jhux2** [comment]
Some more information:  a debug build with `-O0`  and without Valgrind runs to completion without error.  Here's the test I'm running:
```
mpirun -np 6 /scratch/jhu/nalu/nalu/build-gcc-7.2.0-openmpi-1.10.2-debug/naluXd -i /scratch/jhu/nalu/nalu/Nalu/reg_tests/test_files/oversetFluidsEdge/oversetFluidsEdge.i -o oversetFluidsEdge-2.log
```

**Comment by jrood-nrel** [comment]
@jhux2 I also saw the error disappear when using the `Debug` build type. https://github.com/NaluCFD/Nalu/issues/294#issuecomment-357362951

**Comment by jhux2** [comment]
I'm making progress. I got a stack trace by building Trilinos in RELWITHDEBINFO mode (`-O2 -g`, I think):
```
(gdb) where
#0  __cxa_throw ()
    at /sierra/sntools/extras/build/gcc-7.2.0-src/libstdc++-v3/libsupc++/eh_throw.cc:75
#1  0x00000000040992c0 in Kokkos::Impl::throw_runtime_exception(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) ()
    at /scratch/jhu/nalu/trilinos/Trilinos/packages/kokkos/core/src/impl/Kokkos_Error.cpp:72
#2  0x0000000004099d0d in Kokkos::HostSpace::allocate(unsigned long) const ()
    at /scratch/jhu/nalu/trilinos/Trilinos/packages/kokkos/core/src/impl/Kokkos_HostSpace.cpp:250
#3  0x000000000409a2c6 in Kokkos::Impl::SharedAllocationRecord<Kokkos::HostSpace, void>::SharedAllocationRecord(Kokkos::HostSpace const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned long, void (*)(Kokkos::Impl::SharedAllocationRecord<void, void>*)) ()
    at /scratch/jhu/nalu/trilinos/Trilinos/packages/kokkos/core/src/impl/Kokkos_HostSpace.cpp:336
#4  0x0000000001fc9969 in _ZN6Kokkos4ViewIPiJNS_10LayoutLeftENS_6DeviceINS_6OpenMPENS_9HostSpaceEEENS_12MemoryTraitsILj0EEEEEC2IJNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEEERKNS_4Impl12ViewCtorPropIJDpT_EEERKNSt9enable_ifIXntsrSL_11has_pointerES2_E4typeE ()
    at /scratch/jhu/nalu/trilinos/Trilinos/packages/kokkos/core/src/impl/Kokkos_SharedAlloc.hpp:214
#5  0x000000000306b22c in _ZN6Kokkos4ViewIPiJNS_10LayoutLeftENS_6DeviceINS_6Open---Type <return> to continue, or q <return> to quit---
MPENS_9HostSpaceEEENS_12MemoryTraitsILj0EEEEEC2IA8_cEERKT_NSt9enable_ifIXsrNS_4Impl13is_view_labelISC_EE5valueEKmE4typeEmmmmmmm ()
    at /scratch/jhu/nalu/trilinos/Trilinos/packages/kokkos/core/src/Kokkos_View.hpp:1317
#6  0x00000000030b7c7a in _ZN6Tpetra9MMdetails14KernelWrappersIdilN6Kokkos6Compat23KokkosDeviceWrapperNodeINS2_6OpenMPENS2_9HostSpaceEEENS2_4ViewIPiJNS2_10LayoutLeftENS2_6DeviceIS5_S6_EEEEEE33mult_A_B_newmatrix_kernel_wrapperERNS_15CrsMatrixStructIdilS7_EESH_RKSD_SJ_SJ_SJ_RNS_9CrsMatrixIdilS7_Lb0EEEN7Teuchos3RCPIKNS_6ImportIilS7_EEEERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERKNSO_INSN_13ParameterListEEE ()
    at /scratch/jhu/nalu/trilinos/Trilinos/packages/tpetra/core/ext/TpetraExt_MatrixMatrix_def.hpp:2657
#7  0x00000000030bafc2 in void Tpetra::MMdetails::mult_A_B_newmatrix<double, int, long, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::OpenMP, Kokkos::HostSpace> >(Tpetra::CrsMatrixStruct<double, int, long, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::OpenMP, Kokkos::HostSpace> >&, Tpetra::CrsMatrixStruct<double, int, long, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::OpenMP, Kokkos::HostSpace> >&, Tpetra::CrsMatrix<double, int, long, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::OpenMP, Kokkos::HostSpace>::classic>&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, Teuchos::RCP<Teuchos::ParameterList> const&) ()
    at /scratch/jhu/nalu/trilinos/Trilinos/packages/tpetra/core/ext/TpetraExt_Ma---Type <return> to continue, or q <return> to quit---
trixMatrix_def.hpp:2308
#8  0x00000000031552fe in void Tpetra::MatrixMatrix::Multiply<double, int, long, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::OpenMP, Kokkos::HostSpace> >(Tpetra::CrsMatrix<double, int, long, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::OpenMP, Kokkos::HostSpace>::classic> const&, bool, Tpetra::CrsMatrix<double, int, long, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::OpenMP, Kokkos::HostSpace>::classic> const&, bool, Tpetra::CrsMatrix<double, int, long, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::OpenMP, Kokkos::HostSpace>::classic>&, bool, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, Teuchos::RCP<Teuchos::ParameterList> const&) ()
    at /scratch/jhu/nalu/trilinos/Trilinos/packages/tpetra/core/ext/TpetraExt_MatrixMatrix_def.hpp:236
#9  0x0000000001ecef2c in Xpetra::MatrixMatrix<double, int, long, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::OpenMP, Kokkos::HostSpace> >::Multiply(Xpetra::Matrix<double, int, long, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::OpenMP, Kokkos::HostSpace> > const&, bool, Xpetra::Matrix<double, int, long, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::OpenMP, Kokkos::HostSpace> > const&, bool, Xpetra::Matrix<double, int, long, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::OpenMP, Kokkos::HostSpace> >&, bool, bool, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, Teuchos::RCP<Teuchos::Pa---Type <return> to continue, or q <return> to quit---
rameterList> const&) ()
    at /scratch/jhu/nalu/trilinos/Trilinos/packages/xpetra/sup/Utils/Xpetra_MatrixMatrix.hpp:283
#10 0x00000000020cde60 in Xpetra::MatrixMatrix<double, int, long, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::OpenMP, Kokkos::HostSpace> >::Multiply(Xpetra::Matrix<double, int, long, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::OpenMP, Kokkos::HostSpace> > const&, bool, Xpetra::Matrix<double, int, long, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::OpenMP, Kokkos::HostSpace> > const&, bool, Teuchos::RCP<Xpetra::Matrix<double, int, long, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::OpenMP, Kokkos::HostSpace> > >, Teuchos::basic_FancyOStream<char, std::char_traits<char> >&, bool, bool, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, Teuchos::RCP<Teuchos::ParameterList> const&) ()
    at /scratch/jhu/nalu/trilinos/Trilinos/packages/xpetra/sup/Utils/Xpetra_MatrixMatrix.hpp:366
#11 0x00000000021d8a25 in MueLu::RAPFactory<double, int, long, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::OpenMP, Kokkos::HostSpace> >::Build(MueLu::Level&, MueLu::Level&) const ()
    at /scratch/jhu/nalu/trilinos/Trilinos/packages/muelu/src/Misc/MueLu_RAPFactory_def.hpp:155
#12 0x0000000001ab74c6 in MueLu::TwoLevelFactoryBase::CallBuild(MueLu::Level&) const ()
    at /scratch/jhu/nalu/trilinos/Trilinos/packages/muelu/src/Interface/../MueCe---Type <return> to continue, or q <return> to quit---
ntral/MueLu_TwoLevelFactoryBase.hpp:151
#13 0x00000000014abcf2 in Teuchos::RCP<Xpetra::Matrix<double, int, long, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::OpenMP, Kokkos::HostSpace> > >& MueLu::Level::Get<Teuchos::RCP<Xpetra::Matrix<double, int, long, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::OpenMP, Kokkos::HostSpace> > > >(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, MueLu::FactoryBase const*) ()
    at /scratch/jhu/nalu/trilinos/install/relwithdebinfo-gcc-7.2.0_openmpi-1.10.2/include/MueLu_Level.hpp:203
#14 0x00000000022b17a3 in MueLu::RepartitionFactory<double, int, long, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::OpenMP, Kokkos::HostSpace> >::Build(MueLu::Level&) const ()
    at /scratch/jhu/nalu/trilinos/Trilinos/packages/muelu/src/MueCentral/MueLu_Factory.hpp:156
#15 0x0000000001ab7031 in MueLu::SingleLevelFactoryBase::CallBuild(MueLu::Level&) const ()
    at /scratch/jhu/nalu/trilinos/Trilinos/packages/muelu/src/Interface/../MueCentral/MueLu_SingleLevelFactoryBase.hpp:133
#16 0x00000000021052e9 in Teuchos::RCP<Xpetra::Import<int, long, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::OpenMP, Kokkos::HostSpace> > const>& MueLu::Level::Get<Teuchos::RCP<Xpetra::Import<int, long, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::OpenMP, Kokkos::HostSpace> > const> >(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, MueLu::FactoryBase ---Type <return> to continue, or q <return> to quit---
const*) ()
    at /scratch/jhu/nalu/trilinos/Trilinos/packages/muelu/src/MueCentral/MueLu_Level.hpp:203
#17 0x0000000002257470 in MueLu::RebalanceTransferFactory<double, int, long, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::OpenMP, Kokkos::HostSpace> >::Build(MueLu::Level&, MueLu::Level&) const ()
    at /scratch/jhu/nalu/trilinos/Trilinos/packages/muelu/src/MueCentral/MueLu_Factory.hpp:156
#18 0x0000000001ab74c6 in MueLu::TwoLevelFactoryBase::CallBuild(MueLu::Level&) const ()
    at /scratch/jhu/nalu/trilinos/Trilinos/packages/muelu/src/Interface/../MueCentral/MueLu_TwoLevelFactoryBase.hpp:151
#19 0x00000000014a33c2 in Teuchos::RCP<Xpetra::Operator<double, int, long, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::OpenMP, Kokkos::HostSpace> > >& MueLu::Level::Get<Teuchos::RCP<Xpetra::Operator<double, int, long, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::OpenMP, Kokkos::HostSpace> > > >(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, MueLu::FactoryBase const*) ()
    at /scratch/jhu/nalu/trilinos/install/relwithdebinfo-gcc-7.2.0_openmpi-1.10.2/include/MueLu_Level.hpp:203
#20 0x000000000238f9ac in MueLu::TopRAPFactory<double, int, long, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::OpenMP, Kokkos::HostSpace> >::Build(MueLu::Level&, MueLu::Level&) const ()
---Type <return> to continue, or q <return> to quit---q

```

**Comment by jhux2** [comment]
It appears that compiler optimization might have dropped symbols out.
```
#6  0x00000000030b7c7a in _ZN6Tpetra9MMdetails14KernelWrappersIdilN6Kokkos6Compat23KokkosDeviceWrapperNodeINS2_6OpenMPENS2_9HostSpaceEEENS2_4ViewIPiJNS2_10LayoutLeftENS2_6DeviceIS5_S6_EEEEEE33mult_A_B_newmatrix_kernel_wrapperERNS_15CrsMatrixStructIdilS7_EESH_RKSD_SJ_SJ_SJ_RNS_9CrsMatrixIdilS7_Lb0EEEN7Teuchos3RCPIKNS_6ImportIilS7_EEEERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERKNSO_INSN_13ParameterListEEE ()
    at /scratch/jhu/nalu/trilinos/Trilinos/packages/tpetra/core/ext/TpetraExt_MatrixMatrix_def.hpp:2657
2657          lno_nnz_view_t Mcolind("Mcolind",merge_nnz);
(gdb) list
2652                   Mrowptr(i+1)=update;
2653          });
2654
2655          // Allocate nnz
2656          size_t merge_nnz = Mrowptr(merge_numrows);
2657          lno_nnz_view_t Mcolind("Mcolind",merge_nnz);
2658          scalar_view_t Mvalues("Mvals",merge_nnz);
2659
2660          // Use a Kokkos::parallel_for to fill the rowptr/colind arrays
2661          //
(gdb) p merge_nnz
No symbol "merge_nnz" in current context.
```

@mhoemmen Are there any options I can pass to Kokkos via cmake or CMAKE_CXX_FLAGS to get additional debugging checks?

**Comment by jhux2** [comment]
@csiefer2

**Comment by mhoemmen** [comment]
@jhux2 Kokkos should do bounds checking by default in a debug build of Trilinos.  Otherwise, that's pretty much it.  Try setting the TPETRA_DEBUG environment variable in a release build for some extra checks too.

**Comment by jrood-nrel** [comment]
I have enabled OpenMP with GCC 7.2.0 and I am **not** seeing segfaults anymore. As first reported in https://github.com/NaluCFD/Nalu/issues/295 . I think I am considering this specific issue resolved.

**Comment by mhoemmen** [comment]
The fix for #2331 may have fixed this (thanks @crtrott ! ).  I'll close this issue.  Thanks for reporting! :D
